src/main/webapp/app/exercises/shared/exercise-headers/header-participation-page.component.ts

```diff
@@ -33,6 +33,20 @@ export class HeaderParticipationPageComponent implements OnInit, OnChanges {
         this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
     }
 
+    /**
+     * Returns false if it is an exam exercise and the publishResultsDate is in the future, true otherwise
+     */
+    get isPublished(): boolean {
```

Comment:
This is the second occurrence of `isPublished` in the client. You might want to move this to the model or the service.

Reply:
Refactored into the exercise model